server/rpc.go

```diff
@@ -291,12 +314,31 @@ func (h *lphttp) ServeSegment(w http.ResponseWriter, r *http.Request) {
 		glog.Error("Could not transcode ", err)
 		result = net.TranscodeResult{Result: &net.TranscodeResult_Error{Error: err.Error()}}
 	} else {
-		segments := make([]*net.TranscodedSegmentData, len(res.Urls))
-		for i, v := range res.Urls {
-			d := &net.TranscodedSegmentData{
-				Url: orch.ServiceURI().String() + "/stream/" + v,
-			}
+		segments := make([]*net.TranscodedSegmentData, len(res.Data))
+		jid := job.JobId.Int64()
+		for i, data := range res.Data {
+			streamID := res.StreamIDs[i]
+
+			d := &net.TranscodedSegmentData{}
 			segments[i] = d
+			var outputStorage drivers.OSDriver = drivers.LocalStorage
+			if len(segData.ProposedOS) > 0 {
+				outputStorage = drivers.NewDriver(segData.ProposedOS[0].Storage, segData.ProposedOS[0])
+			} else if drivers.ExternalStorage != nil {
+				outputStorage = drivers.ExternalStorage
+			}
+			glog.V(common.DEBUG).Infof("output storage driver is %+v", outputStorage)
+
+			name := fmt.Sprintf("%s_%v.ts", streamID, segData.Seq)
+			start := time.Now()
+			sturi, err := outputStorage.SaveData(jid, streamID, name, data)
+			glog.V(common.DEBUG).Infof("Saving %s to %s took  %s", name, outputStorage.GetInfo().Storage, time.Since(start))
```

Comment:
For some reason the function execution never gets past this log message; the response never gets sent back to the broadcaster. Removing this log message makes things work.

Reply:
This is really really strange. Couple of days ago, when I was testing, S3 started to be very slow - sometime it took 30 seconds to write 100k of data. At first I was thinking that it is not working, and that's why I've added this log message. Maybe this was the case?